internal/extender/failover.go

```diff
@@ -107,8 +107,13 @@ func (r *reconciler) syncResourceReservations(ctx context.Context, sp *sparkPods
 			return
 		}
 
-		podsWithRR := make(map[string]bool, len(rr.Status.Pods))
-		for _, podName := range rr.Status.Pods {
+		newRR, ok := r.resourceReservations.Get(exec.Namespace, sp.appID)
```

Comment:
I think you can return the patched resource reservation from the call to `patchResourceReservation` above, that way you don't have to look up from the cache here

Reply:
👍 Done